### ISSUE TITLE
Automatic correction of `beginning_of_month..end_of_month` fails

### ISSUE THREAD

**Issue by shimadama** [rubocop/rubocop#9918]
With the following code:

```ruby
beginning_of_month..end_of_month
```

If you run the command `bundle exec rubocop -A`, you will get the following error message.

```bash
An error occurred while Rails/ExpandedDateRange cop was inspecting /hoge.rb:48:4.
To see the complete backtrace run rubocop -d.
```

--------

## Expected behavior

It will be fixed as follows
```ruby
all_month
```

## Actual behavior

When I executed the command `rubocop --debug`, I got the following message.

```bash
1 error occurred:
An error occurred while Rails/ExpandedDateRange cop was inspecting /hoge.rb:48:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.14.0 (using Parser 3.0.1.1, rubocop-ast 1.7.0, running on ruby 2.7.1 x86_64-darwin19)
```

## Steps to reproduce the problem

- Create Ruby file containing:
```ruby
def hoge
  beginning_of_month..end_of_month
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop -V
1.18.2 (using Parser 3.0.1.1, rubocop-ast 1.7.0, running on ruby 2.7.1 x86_64-darwin19)
  - rubocop-performance 1.11.3
  - rubocop-rails 2.11.2
  - rubocop-rspec 2.4.0
```
